src/backend/aspen/api/views/samples.py

```diff
@@ -20,12 +24,35 @@ async def list_samples(
     return False
 
 
-@router.delete("/{sample_id}")
+async def get_owned_sample_by_id(db, group_id, sample_id, user):
+    query = sa.select(Sample).filter(
+        sa.and_(
+            Sample.submitting_group == user.group,  # This is an access control check!
+            Sample.submitting_group_id
+            == group_id,  # This makes sure we included the correct group ID in our path.
+            Sample.public_identifier == sample_id,
+        )
+    )
+    results = await db.execute(query)
+    try:
+        return results.scalars().one()
+    except NoResultFound:
+        raise ex.NotFoundException("sample not found")
+
+
+@router.delete("/{group_id}/{sample_id:path}")
```

Comment:
I'm guessing that we may want to allow users that belong to groups other than the owner group (for example: admin users) to be able to delete these samples in the future. Since sample public id's are only required to be unique within a given group, we need to include the group id to uniquely identify a sample.

Reply:
I think the above explanation about why we need `:path` should definitely be a comment in code.

Seems a shame we have to do it this way, but I agree that it's necessary. Hopefully we won't have any sub-attributes under a sample object we will eventually want to delete via CRUD, but that seems reasonably safe.

Reply:
Yeah... I don't have a problem switching to using PK's instead of public ID's here, but if we're going to stick to public identifiers, I'm afraid I don't know of many workarounds.

Reply:
On the whole, I think my vote is to eventually move over everything to PKs as the reference for API calls. It will definitely be a hassle, but the aspect of unambiguous references is so dang useful. That said, devil's advocate, the `/api/samples/validate-ids` route sure has a nice interface with just sending over a list of `sample_ids` and it being inclusive of private, public, AND gisaid. But if we ever have an ambiguous private ID that has been reused over multiple groups, things get messy quickly.

All that said, I do agree what's here in code right now with the whole path suffix resource is totally the right way to go about things if we want to stick with keeping public ID our primary way of referring to a sample in API calls. Even if we're sure we want to switch to PKs in the near future, I think this is totally the right way to go in the meantime.

... Hmmm, totally different alternative that just occurred to me. We _could_ go back to Public ID being fully unique. We can't really make Private ID unique since it should be confidential within a group, but I'd think that Public ID has less of a clear obligation to being totally confidential. Just being aware that _someone_ else took a public ID is an information leak, but it's pretty mild. Maybe we could revert that back to having a uniqueness requirement. Still have the issue with public IDs having `/` and other weird chars for URLs, but hey, we can at least trust it to be unambiguous again. And we could always handle the `/` aspect with URL encoding/decoding if we want to start sub-pathing the resource. ... Then again, if other resources will necessitate PKs anyway, probably not worth it to make samples the only one not using PKs. Just thinking out loud.

Reply:
Ok, based on this feedback, and to reduce confusion. I'm going to update the **write** endpoints to be pk-only for now. I think we also make our read endpoints PK by default, but probably with some options to search via different fields (as needed).